From b141a4bf62ad13d35a61d8747d9121232c1b97df Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Fri, 27 Dec 2024 19:50:07 +0200
Subject: [PATCH] tests: Make WPS protocol tests for credential processing more
 robust

Clear the scan cache at the beginning of these test cases to avoid
issues with WPS_PBC and WPS_PIN commands if there were to be an old scan
result from a previous test case with the same BSSID.

Signed-off-by: Jouni Malinen <j@w1.fi>
---
 tests/hwsim/test_ap_wps.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/hwsim/test_ap_wps.py b/tests/hwsim/test_ap_wps.py
index 883a735e4..c3d0fa4c4 100644
--- a/tests/hwsim/test_ap_wps.py
+++ b/tests/hwsim/test_ap_wps.py
@@ -5586,6 +5586,8 @@ def wps_ext_eap_wsc(dst, src, src_addr, msg):
         raise Exception("EAPOL_RX failed")
 
 def wps_start_ext(apdev, dev, pbc=False, pin=None):
+    dev.flush_scan_cache()
+
     addr = dev.own_addr()
     bssid = apdev['bssid']
     ssid = "test-wps-conf"
-- 
2.43.0

